activerecord/test/cases/primary_class_test.rb

```diff
@@ -110,6 +110,7 @@ def test_application_record_shares_a_connection_with_active_record_by_default
       assert_predicate ApplicationRecord, :application_record_class?
       assert_equal ActiveRecord::Base.connection, ApplicationRecord.connection
     ensure
+      ApplicationRecord.remove_connection
```

Comment:
Rails apps treat `ApplicationRecord` OR the class named `primary_abstract_class` (example below) the same as `ActiveRecord::Base` in order to avoid opening duplicate connections to the same database. In a real app we want these connections to be shared and if `ApplicationRecord.establish_connection` is called it shouldn't replace  or duplicate `ActiveRecord::Base`.

However, in the AR tests we don't actually want `ApplicationRecord` shared with `ActiveRecord::Base` beyond these tests so we need to remove the connection when the test is complete. Previously `establish_connection` was handling this for us, but now we need to do it explicitly. In an app these would actually be shared, but in our tests we don't want to share this global state.